plugins/techdocs/src/reader/components/TechDocsReaderPage/context.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactNode, memo, useState } from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+import { useApi } from '@backstage/core-plugin-api';
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import {
+  TechDocsReaderPageValue,
+  defaultTechDocsReaderPageValue,
+  TechDocsReaderPageContext,
+  useTechDocsReaderPage,
+} from '@backstage/plugin-techdocs-react';
+import { createVersionedValueMap } from '@backstage/version-bridge';
+
+import { techdocsApiRef } from '../../../api';
+
+const areEntityNamesEqual = (
+  prevEntityName: CompoundEntityRef,
+  nextEntityName: CompoundEntityRef,
+) => {
```

Comment:
could do `stringifyEntityRef(a) === stringifyEntityRef(b)`, shorter and handles casing differences